language/move-lang/src/shared/unique_set.rs

```diff
@@ -33,6 +33,44 @@ impl<T: TName> UniqueSet<T> {
         self.0.contains_key_(&x_)
     }
 
+    pub fn map<T2: TName, F>(self, mut f: F) -> UniqueSet<T2>
+    where
+        F: FnMut(T) -> T2,
+    {
+        let underlying_map = self.0;
+        UniqueSet(UniqueMap(
+            underlying_map
+                .0
+                .into_iter()
+                .map(|(t_, (loc, _))| {
+                    let t2 = f(T::add_loc(loc, t_));
+                    let (loc2, t2_) = t2.drop_loc();
+                    (t2_, (loc2, ()))
+                })
+                .collect(),
+        ))
+    }
+
+    pub fn filter_map<T2: TName, F>(self, mut f: F) -> UniqueSet<T2>
+    where
+        F: FnMut(T) -> Option<T2>,
+    {
+        let underlying_map = self.0;
+        UniqueSet(UniqueMap(
+            underlying_map
+                .0
+                .into_iter()
+                .filter_map(|(t_, (loc, _))| {
+                    let t2_opt = f(T::add_loc(loc, t_));
+                    t2_opt.map(|t2| {
+                        let (loc2, t2_) = t2.drop_loc();
+                        (t2_, (loc2, ()))
+                    })
+                })
+                .collect(),
+        ))
+    }
+
```

Comment:
These cannot exist in the current form because you might have collisions or duplicates without errors, and thus they aren't so "unique"

Reply:
I would delete these completely, its for the same reason that rekey isn't a function on uniquemap

Reply:
As discussed with Todd offline, it is not wise to convert friends into using `UniqueSet` for the reason that we need the location for the whole `friend 0x2::A` as well as  the location for `0x2::A` in the error messages. So, the whole commit is dropped.